com.unity.multiplayer.mlapi/Prototyping/NetworkTransform.cs

```diff
@@ -11,364 +11,217 @@ namespace MLAPI.Prototyping
     [AddComponentMenu("MLAPI/NetworkTransform")]
     public class NetworkTransform : NetworkBehaviour
     {
-        internal class ClientSendInfo
+        public enum Authority
         {
-            public float LastSent;
-            public Vector3? LastMissedPosition;
-            public Quaternion? LastMissedRotation;
+            Server = 0, // default
+            Client,
+            Shared
         }
 
+        [SerializeField, Tooltip("Defines who can update this transform.")]
+        public Authority authority; // todo Luke mentioned an incoming system to manage this at the NetworkBehaviour level, lets sync on this
+
         /// <summary>
         /// The base amount of sends per seconds to use when range is disabled
         /// </summary>
         [Range(0, 120)]
-        public float FixedSendsPerSecond = 20f;
-
-        /// <summary>
-        /// Is the sends per second assumed to be the same across all instances
-        /// </summary>
-        [Tooltip("This assumes that the SendsPerSecond is synced across clients")]
-        public bool AssumeSyncedSends = true;
+        public float FixedSendsPerSecond = 30f;
```

Comment:
Why do we need to change it?
Or, more generally, why do we still have this?

It only seems to affect `v.Settings.SendTickrate = FixedSendsPerSecond;`

And the `SendTickrate` is only used by `NetworkList`, `NetworkSet` and `NetworkDictionary`. As such, I'm tempted to say: let's remove FixedSendsPerSecond for NetworkTransform and simply leave `v.Settings.SendTickrate` at default value. If it breaks anything, then I missed something :-)

Reply:
+ 1 for this. Maybe we want `NetworkTransform` to expose a `SendTickrate` property which then updates the position, rotation, scale variable at the same time? Not sure whether that's confusing though and individual control over update rate of for instance scale could be nice.

Reply:
To me, the fact only list set and dict are using SentTickRate is a problem by itself. I wasn't aware of that and that should be fixed.
A use case I can see for this is this:
if I have a far away object, we could have a relevancy system that reduces global tick rate. However, if I have a giant far away, I DO want to see that giant's detailed updates. I would want to override that giant's tick rate to keep it at this value.
Not sure I see a use case for having different control per variable though. Maybe if I know my scale will change in a linear way, I could save on bandwidth and have it update once per second for example and just interpolate it?

Reply:
If there's a valid use case for the user to set properties about priority, importance or desired rates, the user should do it at the NetworkVariable level, and Snapshot should accommodate/consider those values. Then, NetworkTransform can expose those underlying value. My point was that NetworkTransform should not attempt to *drive* this rate, as it kinda did in the past.

Reply:
yep, that's why I'm still exposing that value but not using it. I didn't want to remove too much functionality from the old NetworkTransform.
Once we have that feature available at the NetVar level, we could plug it in, what do you think?